spec/user_interface_spec.rb

```diff
@@ -72,4 +72,61 @@
 
     expect(ui.run).to eq(1)
   end
+
+  describe 'ask for fields method' do
+    let(:input) { StringIO.new(contact_inputs(TEST_DETAILS)) }
+    let(:ui) { described_class.new(input, output) }
+
+    it 'asks user for all fields' do
+      ui.ask_for_fields
+
+      expect(output.string).to include(described_class::FIELDS_TO_PROMPTS.values.join("\n"))
+    end
+
+    it 'reads the input for all prompt' do
+      user_input = ui.ask_for_fields
```

Comment:
this is not the `user_input`, this is the actual `contact_details`. The contact details is what we want this method to return.

I will then also rename the test description to document that this method gets the contact details, as that is what this test is checking :+1: 

```ruby
it 'gets the contact details' do
```